### PR TITLE
[Instrumentation.SqlClient] Move package from main repository

### PR DESCRIPTION
**Pull request by Kielek**
Towards https://github.com/open-telemetry/opentelemetry-dotnet/issues/5526

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

**Comment by Kielek** [comment]
Open questions
1. CHANGELOG.md - should it contain information about moving from main repository to the contib?
2. What's about component owner file? Should it be unset as for other packages migrated from main repository or do you want to be assugned?

@open-telemetry/dotnet-approvers?

**Comment by reyang** [comment]
> 1. CHANGELOG.md - should it contain information about moving from main repository to the contib?

Probably not necessary IMHO. I think including the [opentelemetry-dotnet PR link](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5559) in the commit description is important though.

> 2. What's about component owner file? Should it be unset as for other packages migrated from main repository or do you want to be assugned?

Not a blocker, leave it as empty for now, folks who have knowledge/experience + are willing to be the owner can claim it in separate PRs.

File: test/OpenTelemetry.Instrumentation.SqlClient.Tests/OpenTelemetry.Instrumentation.SqlClient.Tests.csproj [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <Description>Unit test project for OpenTelemetry SqlClient instrumentations</Description>
+    <TargetFrameworks>net8.0;net7.0;net6.0</TargetFrameworks>
```
Comment (vishweshbankwar) [inline test/OpenTelemetry.Instrumentation.SqlClient.Tests/OpenTelemetry.Instrumentation.SqlClient.Tests.csproj:4]: This does not match the original list which has netstandard2.0 as well.

Comment (vishweshbankwar) [reply test/OpenTelemetry.Instrumentation.SqlClient.Tests/OpenTelemetry.Instrumentation.SqlClient.Tests.csproj:4]: My bad, This is test project.

File: src/OpenTelemetry.Instrumentation.SqlClient/OpenTelemetry.Instrumentation.SqlClient.csproj [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    <Compile Include="$(RepoRoot)\src\Shared\ExceptionExtensions.cs" Link="Includes\ExceptionExtensions.cs" />
+    <Compile Include="$(RepoRoot)\src\Shared\Guard.cs" Link="Includes\Guard.cs" />
+    <Compile Include="$(RepoRoot)\src\Shared\ListenerHandler.cs" Link="Includes\ListenerHandler.cs" />
+    <Compile Include="$(RepoRoot)\src\Shared\NullableAttributes.cs" Link="Includes\NullableAttributes.cs" />
+    <Compile Include="$(RepoRoot)\src\Shared\PropertyFetcher.AOT.cs" Link="Includes\PropertyFetcher.AOT.cs" />
+    <Compile Include="$(RepoRoot)\src\Shared\SemanticConventions.cs" Link="Includes\SemanticConventions.cs" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Extensions.Options" Version="$(MicrosoftExtensionsOptionsPkgVer)" />
```
Comment (vishweshbankwar) [inline src/OpenTelemetry.Instrumentation.SqlClient/OpenTelemetry.Instrumentation.SqlClient.csproj:35]: This is downgrading the version to 3.1.0?

File: test/OpenTelemetry.Contrib.Tests.Shared/EnabledOnDockerPlatformTheoryAttribute.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma warning disable IDE0005 // Using directive is unnecessary.using System;
```
[the remark this answers is not in the document]
Comment (Kielek) [reply test/OpenTelemetry.Contrib.Tests.Shared/EnabledOnDockerPlatformTheoryAttribute.cs:4]: Upstream - all usages have enable `ImplicitUsings`. In this repository, this file is also part of `OpenTelemetry.Contrib.Tests.Shared`. This feature is not enabled so I have to add it manually, but the SqlClient project enables `Implicitusings` - it requiers additional #pragma disable.

File: src/OpenTelemetry.Instrumentation.SqlClient/OpenTelemetry.Instrumentation.SqlClient.csproj (hunk 2 above)
Comment (Kielek) [reply src/OpenTelemetry.Instrumentation.SqlClient/OpenTelemetry.Instrumentation.SqlClient.csproj]: Good catch.  Handled in c2fd7e00f74c993ab9dfb81ae819394369f00108.

We should consider unifying `Microsoft.Extensions.Options` version in this repository. Candidate for separate PR.

**Review by vishweshbankwar — APPROVED** [review]

**Pull request merged**